test/assert.js

```diff
@@ -1998,13 +2088,44 @@ describe('assert', function () {
 
     err(function() {
       assert.isAtMost(null, 1, 'blah');
-    }, 'blah: expected null to be a number');
+    }, 'blah: expected null to be a number or a date');
 
     err(function() {
       assert.isAtMost(1, null, 'blah');
     }, 'blah: the argument to most must be a number');
   });
 
+  it('atMost (dates)', function() {
+    var now = new Date();
+    var oneSecondAgo = new Date(now.getTime() - 1);
+    var oneSecondAfter = new Date(now.getTime() + 1);
+
+    assert.isAtMost(oneSecondAgo, now, 'Now should be below one second ago');
+    assert.isAtMost(now, now, 'Now should be equal to now');
+
+    err(function() {
+      assert.isAtMost(oneSecondAfter, now, 'blah');
+    }, 'blah: expected ' + oneSecondAgo.toUTCString() + ' to be at most ' + now.toUTCString());
+
+    err(function() {
+      assert.isAtMost(null, now, 'blah');
+    }, 'blah: expected null to be a number or a date');
+
+    err(function() {
+      assert.isAtMost(now, null, 'blah');
+    }, 'blah: the argument to most must be a date');
+
+    err(function() {
+      assert.isAtMost(now, 1, 'blah');
+    }, 'blah: type mismatch, expected to most value to be a date');
+
+    err(function() {
+      assert.isAtMost(1, now, 'blah');
+    }, 'blah: type mismatch, expected to most value to be a number');
+  });
+
+  // There's no assert.isWithin(n, a, b)?
```

Comment:
While should an expect implement have this options, it looks like there's no simple assert.isWithin()
Should we add an alias?